File: src/main/java/app/UserController.java
package app;

import java.util.HashMap;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;

public class UserController {
    private Map<String, String> users;

    public UserController() {
        users = new HashMap<>();
    }

    public boolean registerUser(String username, String password) {
        if (username == null || password == null || username.isEmpty() || password.isEmpty()) {
            throw new IllegalArgumentException("Username and password required");
        }
        if (users.containsKey(username)) {
            return false;
        }
        users.put(username, password);
        return true;
    }

    public boolean loginUser(String username, String password) {
        if (username == null || password == null || username.isEmpty() || password.isEmpty()) {
            throw new IllegalArgumentException("Missing credentials");
        }
        return password.equals(users.get(username));
    }
}

File: app/user_controller.py
class UserController:
    def __init__(self):
        self.users = {}  # username: password

    def register_user(self, username, password):
        if not username or not password:
            raise ValueError("Username and password required")
        if username in self.users:
            return False
        self.users[username] = password
        return True

    def login_user(self, username, password):
        if not username or not password:
            raise ValueError("Missing credentials") 
        return self.users.get(username) == password

    def get_user(self, username):
        """Lấy thông tin user theo username"""
        return self.users.get(username)

    def user_exists(self, username):
        """Kiểm tra user có tồn tại không"""
        return username in self.users

    def get_all_users(self):
        """Lấy danh sách tất cả users"""
        return list(self.users.keys())

